src/picongpu/include/plugins/adios/ADIOSWriter.hpp

```diff
@@ -428,41 +446,142 @@ class ADIOSWriter : public ILightweightPlugin
+            mThreadParams.fp = adios_read_open(strFname.str().c_str(),
+                        ADIOS_READ_METHOD_BP, mThreadParams.adiosComm,
+                        ADIOS_LOCKMODE_CURRENT, timeout);
         }
+        if (adios_errno == err_end_of_stream )
+            /* could not read full stream */
+            throw std::runtime_error("ADIOS: Stream terminated too early: " +
+                                     std::string(adios_errmsg()) );
+        if (mThreadParams.fp == NULL)
+            throw std::runtime_error("ADIOS: Error opening stream: " +
+                                     std::string(adios_errmsg()) );
+
+        /* ADIOS types */
+        AdiosUInt32Type adiosUInt32Type;
 
-        for (uint32_t i = 0; i < simDim; ++i)
+        /* load number of slides to initialize MovingWindow */
+        log<picLog::INPUT_OUTPUT > ("ADIOS: (begin) read attr (%1% available)") %
+            mThreadParams.fp->nattrs;
+        void* slidesPtr = NULL;
+        int slideSize;
+        enum ADIOS_DATATYPES slidesType;
+        ADIOS_CMD(adios_get_attr( mThreadParams.fp,
+                                  (mThreadParams.adiosBasePath + std::string("sim_slides")).c_str(),
+                                  &slidesType,
+                                  &slideSize,
+                                  &slidesPtr ));
+
+        uint32_t slides = *( (uint32_t*)slidesPtr );
+        log<picLog::INPUT_OUTPUT > ("ADIOS: value of sim_slides = %1%") %
+            slides;
+
+        assert(slidesType == adiosUInt32Type.type);
+        assert(slideSize == sizeof(uint32_t)); // uint32_t in bytes
+
+        void* lastStepPtr = NULL;
+        int lastStepSize;
+        enum ADIOS_DATATYPES lastStepType;
+        ADIOS_CMD(adios_get_attr( mThreadParams.fp,
+                                  (mThreadParams.adiosBasePath + std::string("iteration")).c_str(),
+                                  &lastStepType,
+                                  &lastStepSize,
+                                  &lastStepPtr ));
+        uint32_t lastStep = *( (uint32_t*)lastStepPtr );
+        log<picLog::INPUT_OUTPUT > ("ADIOS: value of iteration = %1%") %
+            lastStep;
+
+        assert(lastStepType == adiosUInt32Type.type);
+        assert(lastStep == restartStep);
+
+        /* apply slides to set gpus to last/written configuration */
+        log<picLog::INPUT_OUTPUT > ("ADIOS: Setting slide count for moving window to %1%") % slides;
+        MovingWindow::getInstance().setSlideCounter(slides, restartStep);
+
+        /* re-distribute the local offsets in y-direction */
+        GridController<simDim> &gc = Environment<simDim>::get().GridController();
+        if( MovingWindow::getInstance().isSlidingWindowActive() )
+            gc.setStateAfterSlides(slides);
+
+        /* set window for restart, complete global domain */
+        mThreadParams.window = MovingWindow::getInstance().getDomainAsWindow(restartStep);
+        for (uint32_t d = 0; d < simDim; ++d)
```

Comment:
Can we remove this loop and change it to:
` mThreadParams.localWindowToDomainOffset = DataSpace<simDim>::create(0);`